### ISSUE TITLE
fft2 different result in numpy and matlab

### ISSUE THREAD

**Issue by TianyangGai** [numpy/numpy#13424]
I was trying to port one code from python to matlab, but I encounter one inconsistence between numpy fft2 and matlab fft2：

### Reproducing code example:

**the matlab test code:**
matlab 2012b
```matlab
>> u = zeros(500);
>> u(201:300, 201:300)=1;
>> fft2_u = fft2(u);
>> sum(sum(fft2_u))

ans =

  -1.4567e-11 - 5.6133e-13i
```

**the python test code**
Python 3.6 + Numpy 1.15.1
```python
from numpy import *
from numpy.fft import fft2
u = zeros((500, 500))
u[200:300, 200:300] = 1
fft2_u = fft2(u)
print(fft2_u.sum())

>(-5.343281372915953e-12+3.055333763768431e-13j)
```
The result is quite different.

**Comment by bashtage** [comment]
This difference is small considering you are summing over 25000 elements.  It is also platform dependent, I get `(-1.2505552149377763e-12+9.237055564881302e-14j)` in Python and `-2.3269e-11 + 8.5265e-14i` in MATLAB.

**Comment by seberg** [comment]
Yes, it is mostly numerical imprecision. However, there is likely a bigger point here as well, since numpy's FFT was always slightly less precise then it should be. This has been fixed in 1.16.x.

@TianyangGai it thus probably makes sense if you try upgrading numpy. You should not expect identical results, but you should probably get closer results. I will close the issue, since I believe it is addressed as far as possible, but please do not hesitate to keep discussing, and we can reopen it any time.
